mslib/msui/mpl_pathinteractor.py

```diff
@@ -211,25 +213,26 @@ def update_from_WaypointsTableModel(self, wps_model):
         lats, lons, times = wps_model.intermediate_points(
             numpoints=self.numintpoints, connection="greatcircle")
 
-        # Determine indices of waypoints in list of intermediate points.
-        # Store these indices.
-        waypoints = [[wp.lat, wp.lon] for wp in wps_model.all_waypoint_data()]
-        intermediate_indexes = []
-        ipoint = 0
-        for i, (lat, lon) in enumerate(zip(lats, lons)):
-            if abs(lat - waypoints[ipoint][0]) < 1E-10 and abs(lon - waypoints[ipoint][1]) < 1E-10:
-                intermediate_indexes.append(i)
-                ipoint += 1
-            if ipoint >= len(waypoints):
-                break
-
-        self.intermediate_indexes = intermediate_indexes
-        self.ilats = lats
-        self.ilons = lons
-        self.itimes = times
-
-        # Call super method.
-        WaypointsPath.update_from_WaypointsTableModel(self, wps_model)
+        if lats is not None:
```

Comment:
So far the flightrack is required to consist of at least two points. otherwise a lot of stuff might break...

Reply:
> So far the flightrack is required to consist of at least two points. otherwise a lot of stuff might break...

when the flightrack can't be loaded then it is set to None, see mslib/utils.py